File: Assets/CustomScripts/Player/ROB_Interface.js

function Start () {

}

function Update () {

}

public interface ROBInterface{

	function UpdateSmoothedMovementDirection ();
	function ApplyJumping ();
	function ApplyGravity ();
	// Determine the direction of the player against the surface (for climbing side to side and wallsliding)
	function DirectionOnWall(dir);
	function CalculateJumpVerticalSpeed (targetJumpHeight : float);

	// Handle the jump/double-jump actions
	function DidJump ();

	// Player has come in contact with a surface
	function OnControllerColliderHit (hit : ControllerColliderHit );
	
	// These functions just return the truth value of their implied variables
	function GetSpeed (); 
	
	function IsJumping (); 
	
	function IsDoubleJumping (); 

	function IsGrounded ():boolean;

	function GetDirection (); 

	
	function IsMovingBackwards (); 

	
	function GetLockCameraTimer (); 
	function IsMoving (): boolean;
	
	function HasJumpReachedApex ();
	
	function Reset ();
	

}

File: Assets/CustomScripts/Player/ROB_Interface.cs
using System;
using UnityEngine;

namespace AssemblyCSharp
{
	public interface ROBInterface
	{
		void UpdateSmoothedMovementDirection ();
		void ApplyJumping ();
		void ApplyGravity ();
		// Determine the direction of the player against the surface (for climbing side to side and wallsliding)
		Vector3 DirectionOnWall();
		float CalculateJumpVerticalSpeed (float targetJumpHeight);
	
		// Handle the jump/double-jump actions
		void DidJump ();
	
		// Player has come in contact with a surface
		void OnControllerColliderHit (ControllerColliderHit hit);
		
		// These functions just return the truth value of their implied variables
		//function GetSpeed (); 
		
		bool IsJumping (); 
		
		bool IsDoubleJumping (); 
	
		bool IsGrounded ();
	
		//function GetDirection (); 
	
		
		//function IsMovingBackwards (); 
	
		
		//function GetLockCameraTimer (); 
		//function IsMoving (): boolean;
		
		//function HasJumpReachedApex ();
		
		//function Reset ();
	}
}

